eng/pipelines/pool-migration.yml

```diff
@@ -0,0 +1,98 @@
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-ios
+      ExcludePaths: "^./sdk/|^./eng/common/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-java
+      ExcludePaths: "^./sdk/|^./eng/common/|/resources/session-records/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-js
+      ExcludePaths: "^./sdk/|^./eng/common/|/recordings/node/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-net
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/SessionRecords/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-python
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/recordings/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+# The input parameter format is like:
+# - MigrateFrom: azsdk-pool-mms-win-2019-general
+#   MigrateTo: azsdk-pool-mms-win-2022-general
+# - MigrateFrom: windows-2019
+#   MigrateTo: windows-2022
+# - MigrateFrom: windows2019
+#   MigrateTo: windows2022
+- name: MigrationMap
+  type: object
+  default: []
+trigger: none
+pr: none
+
+jobs:
+  - job: Migration
+    pool:
+      name: azsdk-pool-mms-ubuntu-2004-general
+    variables:
+      - template: ./templates/variables/globals.yml
+    steps:
+      - ${{ each repo in parameters.Repos }}:
+        - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+          parameters:
+            SkipCheckoutNone: true
+            Repositories:
+              - Name: ${{ repo.RepoOwner }}/${{ repo.RepoName }}
+                WorkingDirectory: $(Build.SourcesDirectory)/docs/${{ repo.RepoName }}
+            Paths:
+              - /*
+        - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        - pwsh: |
+            $migrationMap = '${{ convertToJson(parameters.MigrationMap) }}'
+            $(Build.SourcesDirectory)/eng/scripts/Repo-File-Content-Replacements.ps1 `
+              -ExcludePathsRegex '${{ repo.ExcludePaths }}' `
+              -IncludePathsRegex '${{ repo.IncludePaths }}' `
+              -MigrationMap $migrationMap
+          displayName: Update pool version for "${{ repo.RepoName }}"
+          workingDirectory: $(Build.SourcesDirectory)/docs/${{ repo.RepoName }}
+          continueOnError: true
+        - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
+          parameters:
+            BaseBranchName: $(DefaultBranch)
+            PRBranchName: migrate-pool
+            CommitMsg: "Migrate to new pool"
+            RepoOwner: ${{ repo.RepoOwner }}
+            RepoName: ${{ repo.RepoName }}
+            PRTitle: "Migrate to new pool"
+            WorkingDirectory: $(Build.SourcesDirectory)/docs/${{ repo.RepoName }}
```

Comment:
I did not find a good way to convert DevOps identity to GitHub alias. Skip the feature of assigning the PR to the one who triggers the pipeline.

Reply:
Should we assign it to the azure sdk engsys team?

Reply:
@weshaggard How do you think of adding azure-sdk-eng to the PR?

Reply:
We add a parameter of `RequestBy`, so we can pass github alias directly to the pipeline.

Try to add "Azure/azure-sdk-eng" by default as team reviewers, but github API did not take affect. 
Ideally, we can convert devops alias (whoever trigger the pipelin) to github alias. Since the pipeline doesn't use quite often, it is not worth complicate the logic here.